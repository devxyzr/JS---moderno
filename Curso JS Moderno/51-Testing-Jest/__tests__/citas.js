import Citas from "../js/classes/Citas";

describe("Probar la clase de Citas", () => {
  const citas = new Citas();

  const id = Date.now();

  test("Agregar una nueva cita", () => {
    const citaObj = {
      id,
      mascota: "Hook",
      propietario: "Juan",
      telefono: "12332123654",
      fecha: "10-12-2020",
      hora: "10:30",
      sintomas: "Solo duerme",
    };

    citaObj.id = Date.now();

    citas.agregarCita(citaObj);

    // Prueba
    expect(citas).toMatchSnapshot();
  });

  test("Actualizar cita", () => {
    const citaActualizada = {
      id,
      mascota: "Nuevo nombre",
      propietario: "Juan",
      telefono: "12332123654",
      fecha: "10-12-2020",
      hora: "10:30",
      sintomas: "Solo duerme",
    };

    citas.editarCita(citaActualizada);

    expect(citas).toMatchSnapshot();
  });

  test("Eliminar Cita", () => {
    citas.eliminarCita(id);

    expect(citas).toMatchSnapshot();
  });
});
