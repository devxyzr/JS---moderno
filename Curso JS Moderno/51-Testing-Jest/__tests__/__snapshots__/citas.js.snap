// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1656547705358,
      "mascota": "Hook",
      "propietario": "Juan",
      "sintomas": "Solo duerme",
      "telefono": "12332123654",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1656547705358,
      "mascota": "Hook",
      "propietario": "Juan",
      "sintomas": "Solo duerme",
      "telefono": "12332123654",
    },
  ],
}
`;
